.github/actions/install-rust-toolchain/action.yml

```
@@ -0,0 +1,35 @@
+## Inspired by: https://github.com/actions-rs/toolchain/issues/216#issuecomment-1291613319
+name: Install Rust toolchain
+description: "Install Rust toolchain, with caching"
+
+inputs:
+  toolchain:
+    required: false
```

We want to enforce a toolchain
```suggestion
    required: true
```